lib/temporal/workflow/context.rb

```diff
@@ -116,6 +116,7 @@ def execute_workflow(workflow_class, *input, **args)
           timeouts: execution_options.timeouts,
           headers: execution_options.headers,
           memo: execution_options.memo,
+          workflow_id_reuse_policy: options[:workflow_id_reuse_policy]
```

Comment:
nit: can you please make it similar to `parent_close_policy`? This is a very similar behaviour, no reason for different implementation